patches/tModLoader/Terraria/GameContent/ObjectInteractions/NPCSmartInteractCandidateProvider.cs.patch

```diff
@@ -0,0 +1,17 @@
+--- src/Terraria/Terraria/GameContent/ObjectInteractions/NPCSmartInteractCandidateProvider.cs
++++ src/tModLoader/Terraria/GameContent/ObjectInteractions/NPCSmartInteractCandidateProvider.cs
+@@ -1,4 +_,5 @@
+ using Microsoft.Xna.Framework;
++using Terraria.ModLoader;
+ 
+ namespace Terraria.GameContent.ObjectInteractions
+ {
+@@ -43,7 +_,7 @@
+ 			float npcDistanceFromCursor = -1f;
+ 			for (int i = 0; i < 200; i++) {
+ 				NPC nPC = Main.npc[i];
+-				if (nPC.active && nPC.townNPC && nPC.Hitbox.Intersects(value) && !flag) {
++				if (nPC.active && NPCLoader.CanChat(nPC, nPC.townNPC) && nPC.Hitbox.Intersects(value) && !flag) {
```

Comment:
The design of this hook is inconsistent with other hooks which return nullables. We should make `NPCLoader.CanChat` return `bool?` and right here just do `(NPCLoader.CanChat(nPC) ?? nPC.townNPC)`